numba_dpex/core/parfors/parfor_lowerer.py

```diff
@@ -68,11 +69,8 @@ def _getvar(lowerer, x):
         var_val = lowerer.varmap[x]
 
     if var_val:
-        if not isinstance(var_val.type, llvmir.PointerType):
-            with lowerer.builder.goto_entry_block():
-                var_val_ptr = lowerer.builder.alloca(var_val.type)
-            lowerer.builder.store(var_val, var_val_ptr)
-            return var_val_ptr
+        if isinstance(var_val.type, llvmir.PointerType):
```

Comment:
Why is this change needed? Is `var_val` always a `PointerType`?

Reply:
It used to return pointer to value, not it returns value as description says.

Reply:
Change needed in parfor to be aligned with the semantic of numba's unboxed llvm values. Resulted llvmir is not affected, since value to ptr concertation is done in another step ( `KernelLauncerIRBuilder.set_arguments` ).

Reply:
It also makes sense because `ParforKernel.kernel_args_ty` now actually match types of the llvm values.